api/v1alpha1/contour_types.go

```diff
@@ -89,6 +89,20 @@ type ContourSpec struct {
 	// +kubebuilder:validation:MaxLength=253
 	// +optional
 	GatewayClassRef *string `json:"gatewayClassRef,omitempty"`
+
+	// IngressClassName is the name of the IngressClass used by Contour. If unset,
+	// Contour will process all ingress objects without an ingress class annotation
+	// or ingress objects with an annotation matching ingress-class=contour. When
+	// specified, Contour will only process ingress objects that match the provided
+	// class.
+	//
+	// For additional IngressClass details, refer to:
+	//   https://projectcontour.io/docs/main/config/annotations/#ingress-class
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	// +optional
+	IngressClassName *string `json:"ingressClassName,omitempty"`
```

Comment:
Are there any more "ingress-y" things vs gatewayapi things we'd want to split apart into structs? We've done this with the "policy" stuff in Contour to allow ways to grow values and put then in categories. 

Just trying to think ahead before we get too far down the road.

Reply:
+1 thats a good point

Reply:
@stevesloka I can't think of anything from an Ingress standpoint. Doing a quick review of the Contour ConfigMap, I don't see anything Ingress-specific that may be exposed through the Contour API. I see Contour has a `ingress-status-address` flag. Is this a commonly used flag?

As for Gateway API, Contour references a GatewayClass and vice versa. A GatewayClass is pretty minimal. Are there any other config knobs that you expect to expose from Contour to support the Gateway API?

Reply:
:+1: in general that adding some nesting to the spec here will probably be helpful long-term as we add more and more toggles. I guess we still have room to move things around given that the API is alpha right now, though.